api/client/build.go

```diff
@@ -308,13 +323,25 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		if jerr.Code == 0 {
 			jerr.Code = 1
 		}
+		if *suppressOutput {
+			fmt.Fprintln(cli.err, outBuf)
```

Comment:
like above, we need to keep the diagnostic output separate from the actual error message so that error processing has just the error text and not any extra stuff.
```
fmt.Fprintln(cli.err, outBuf, sopts.out)
```
and removing the return so it falls thru should do it.